lib/chef/provider/aws_launch_configuration.rb

```diff
@@ -10,22 +10,20 @@ def create_aws_object
     image_id = Chef::Resource::AwsImage.get_aws_object_id(new_resource.image, resource: new_resource)
     instance_type = new_resource.instance_type || new_resource.driver.default_instance_type
     options = AWSResource.lookup_options(new_resource.options || options, resource: new_resource)
+    options[:launch_configuration_name] = new_resource.name if !new_resource.name.nil?
```

Comment:
Instead of `if !new_resource.name.nil?`, `if new_resource.name` can be written. This avoids calling `!` and `nil?`.